nexus/types/src/external_api/views.rs

```diff
@@ -244,7 +244,22 @@ pub struct VpcRouter {
 pub struct IpPool {
     #[serde(flatten)]
     pub identity: IdentityMetadata,
-    pub silo_id: Option<Uuid>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum IpPoolResourceType {
+    Fleet,
+    Silo,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq)]
+pub struct IpPoolResource {
+    // TODO: is including the pool ID redundant? it's convenient to have and
+    // makes this response a cohesive whole
```

Comment:
```suggestion
```

I would keep the pool id. It doesn't hurt to have and it makes the response more useful by itself.